client-runtime/src/main/java/com/microsoft/rest/http/HttpRequest.java

```diff
@@ -103,7 +103,8 @@ public HttpRequest withBody(byte[] body, String mimeType) {
     public HttpRequest withBody(HttpRequestBody body, String mimeType) {
         this.body = body;
         this.mimeType = mimeType;
-        return withHeader("Content-Length", String.valueOf(body.contentLength()));
+        headers.set("Content-Length", String.valueOf(body.contentLength()));
```

Comment:
What was wrong with "withHeader()"?

Reply:
I believe it was that if Content-Length is already set, it needs to be replaced and not appended to.